#                                                #NYC DEATH RATES
                                                KHAIRUL_CHOWDHURY--IS362--PROJECT2
New York City Leading Causes of Death. The data is captured is from https://data.cityofnewyork.us/Health/New-York-City-Leading-Causes-of-Death/jb7j-dtam/data

In [35]:
import pandas as pd # importing pandas module for usage
import numpy as np

deaths = pd.read_csv('NYC-Death-Rates.csv')

deaths

Year                                      Leading Cause Sex  \
0     2010            Influenza (Flu) and Pneumonia (J09-J18)   F   
1     2008  Accidents Except Drug Posioning (V01-X39, X43,...   F   
2     2013  Accidents Except Drug Posioning (V01-X39, X43,...   M   
3     2010          Cerebrovascular Disease (Stroke: I60-I69)   M   
4     2009                 Assault (Homicide: Y87.1, X85-Y09)   M   
...    ...                                                ...  ..   
1089  2012            Influenza (Flu) and Pneumonia (J09-J18)   F   
1090  2014  Accidents Except Drug Posioning (V01-X39, X43,...   F   
1091  2009              Malignant Neoplasms (Cancer: C00-C97)   M   
1092  2009    Intentional Self-Harm (Suicide: X60-X84, Y87.0)   M   
1093  2013  Essential Hypertension and Renal Diseases (I10...   M   

          Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
0               Hispanic    228       18.7                    23.1  
1               Hispanic     68        5.8                     6.6  
2     White Non-Hispanic    271       20.1                    17.9  
3               Hispanic    140       12.3                    21.4  
4     Black Non-Hispanic    255         30                      30  
...                  ...    ...        ...                     ...  
1089  Not Stated/Unknown      6          .                       .  
1090  White Non-Hispanic    169       11.9                     7.4  
1091  White Non-Hispanic   3236      240.5                   205.6  
1092  White Non-Hispanic    191       14.2                      13  
1093  Black Non-Hispanic    148       17.2                    20.9  

[1094 rows x 7 columns]

In [36]:
tidy = deaths.replace(to_replace = '.', value = np.nan)
tidy

Year                                      Leading Cause Sex  \
0     2010            Influenza (Flu) and Pneumonia (J09-J18)   F   
1     2008  Accidents Except Drug Posioning (V01-X39, X43,...   F   
2     2013  Accidents Except Drug Posioning (V01-X39, X43,...   M   
3     2010          Cerebrovascular Disease (Stroke: I60-I69)   M   
4     2009                 Assault (Homicide: Y87.1, X85-Y09)   M   
...    ...                                                ...  ..   
1089  2012            Influenza (Flu) and Pneumonia (J09-J18)   F   
1090  2014  Accidents Except Drug Posioning (V01-X39, X43,...   F   
1091  2009              Malignant Neoplasms (Cancer: C00-C97)   M   
1092  2009    Intentional Self-Harm (Suicide: X60-X84, Y87.0)   M   
1093  2013  Essential Hypertension and Renal Diseases (I10...   M   

          Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
0               Hispanic    228       18.7                    23.1  
1               Hispanic     68        5.8                     6.6  
2     White Non-Hispanic    271       20.1                    17.9  
3               Hispanic    140       12.3                    21.4  
4     Black Non-Hispanic    255         30                      30  
...                  ...    ...        ...                     ...  
1089  Not Stated/Unknown      6        NaN                     NaN  
1090  White Non-Hispanic    169       11.9                     7.4  
1091  White Non-Hispanic   3236      240.5                   205.6  
1092  White Non-Hispanic    191       14.2                      13  
1093  Black Non-Hispanic    148       17.2                    20.9  

[1094 rows x 7 columns]

# Q1. Which year had the most deaths ?

In [38]:
deaths.groupby('Year').count()

Leading Cause  Sex  Race Ethnicity  Deaths  Death Rate  \
Year                                                           
2007            141  141             141     141         141   
2008            136  136             136     136         136   
2009            135  135             135     135         135   
2010            138  138             138     138         138   
2011            141  141             141     141         141   
2012            134  134             134     134         134   
2013            133  133             133     133         133   
2014            136  136             136     136         136   

      Age Adjusted Death Rate  
Year                           
2007                      141  
2008                      136  
2009                      135  
2010                      138  
2011                      141  
2012                      134  
2013                      133  
2014                      136

# Answer Q1. We can see 2011 had the most deaths with a total of 43.

# Q2. Which sex had the highest death amount ?

In [41]:
deaths.groupby('Race Ethnicity')['Sex'].count()

Race Ethnicity
Asian and Pacific Islander    177
Black Non-Hispanic            178
Hispanic                      177
Not Stated/Unknown            200
Other Race/ Ethnicity         186
White Non-Hispanic            176
Name: Sex, dtype: int64

## Answer Q2. We can see the "Not Stated/Unknown" Race has the highest death toll with a total of 200, followed by "Black Non-Hispanic & Hipanics" at a total of 178.

# Q3. What were the top 5 leading causes of deaths in 2007 ?

In [44]:

deaths[deaths['Year']==2007].sort_values('Deaths', ascending = False).head(5)

Year                                      Leading Cause Sex  \
340  2007                                   All Other Causes   M   
490  2007              Malignant Neoplasms (Cancer: C00-C97)   F   
366  2007                        Diabetes Mellitus (E10-E14)   M   
155  2007  Nephritis, Nephrotic Syndrome and Nephrisis (N...   F   
878  2007                               Septicemia (A40-A41)   F   

         Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
340            Hispanic    988       90.6                   142.6  
490            Hispanic    969         83                   100.7  
366  Not Stated/Unknown      9          .                       .  
155  White Non-Hispanic     89        6.2                     3.5  
878  Black Non-Hispanic     89        8.5                     8.1

## Answer Q3. We can see that the top (5) leading causes of death were Malignant Neoplasms (Cancer: C00-C97) (969),both are same Septicemia (89) and Nephritis, Nephrotic Syndrome and Nephrisis  (89), and some other causes (988).